spec/ics-2-consensus-verification/README.md

```diff
@@ -0,0 +1,265 @@
+---
+
+ics: 2
+title: Consensus Verification
+stage: draft
+category: ibc-core
+requires: 23, 24
+required-by: 3
+author: Juwoon Yun <joon@tendermint.com>, Christopher Goes <cwgoes@tendermint.com>
+created: 2019-02-25
+modified: 2019-04-29
+
+---
+
+## Synopsis
+
+This standard specifies the properties that consensus algorithms of chains implementing IBC are
+expected to satisfy. The properties are needed for efficient and safe verification in the higher
+level protocol abstractions. The algorithm which uses these properties to verify substates of
+another chain is referred to as a "light client verifier", and pairing it with a state that the
+verifier have to trust forms a "light client".
+
+This standard also specifics how the light clients will be stored, registered, and updated on a
+blockchain. The stored lightclient instances will be able to be verified by a thrid party actor.
+
+## Specification
+
+### Motivation
+
+Light clients are the verification method of IBC protocol. One chain can track another chain's
+updating state with a light client pointing to that chain. This standard formalises the common
+model of light client to minimise the dependency on consensus algorithms, so the protocol can
+easily connect with new chains which are running new consensus algorithms, without need to
+upgrade the light client protocol itself.
+
+<!--
+`FullNode`s are procedures running a `Consensus`. Given a `([Transaction], Commit)`, a
+`FullNode` can compute the result `ConsensusState` that the `Consensus` is expected to commit on
+with the same `[Transaction]`, if exists.
+--->
+### Desired Properties
+
+This standard specification provides secure layer to verify other chains' canonical headers,
+using the existing `ConsensusState`. The higher level logics can be able to verify the substate
+with the `CommitmentRoot` stored in the `ConsensusState`, which is guaranteed to be committed by
+the other chain's consensus algorithm.
+
+* `Verifier`s are expected to reflect the behaviour of the full node which is running on the  
+corresponding consensus algorithm. Given a `ConsensusState` and `[Message]`, if a full node
+accepts the new `Header` generated with `Commit`, then the light client MUST also accept it,
+and if a full node rejects, then the light client MUST also reject. The consensus algorithm
+ensures this correspondence. However light clients are not replaying the whole messages, so it
```

Comment:
"The consensus algorithm ensures this correspondence" seems wrong, we can't define the correspondence without reasoning about both a consensus algorithm and a light client algorithm

Reply:
still needs to be addressed